src/frontend/src/components/groups/GroupRow/GroupRow.tsx

```diff
@@ -0,0 +1,117 @@
+import { Avatar, Box, Card, CheckBox, Grid, Text } from 'grommet';
+import React, { ChangeEvent, useRef, useState } from 'react';
+import { More } from 'grommet-icons';
+import { useMediaQuery } from 'react-responsive';
+
+export interface Group {
+  id: string;
+  name: string;
+  members: {
+    id: string;
+    name: string;
+    avatar: string;
+  }[];
+}
+
+export interface GroupRowProps {
+  /**
+   * The group to display
+   */
+  group: Group;
+  /**
+   * Callback when the group is selected or deselected
+   */
+  onToogle: (selected: boolean) => void;
+  /**
+   * Is the group currently selected
+   */
+  selected: boolean;
+}
+
+export default function GroupRow(props: GroupRowProps) {
+  const { group, onToogle, selected } = props;
+  const ref = useRef<HTMLInputElement>(null);
```

Comment:
This is not true. The ref contains an `HTMLInputElement` or `null`. 
I suggest you to create some type utils and declare these two you will use a lot:

```
export type Maybe<T> = T | undefined;

export type Nullable<T> = T | null;
```

And then here 

```suggestion
  const ref = useRef<Nullable<HTMLInputElement>>(null);
```

Reply:
Indeed. In this case I need to remove the ref completely as it is useless, but I keep the ideal of such generics. Thank you